pkgs/development/tools/rust/racerd/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, rustPlatform, makeWrapper }:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "racerd-${version}";
+  version = "0.1.1";
+  src = fetchFromGitHub {
+    owner = "jwilm";
+    repo = "racerd";
+    rev = "v${version}";
+    sha256 = "0kggikin8hyq71hwbapczyvik39wwycvgdxigzhgivlcpss2wmdx";
+  };
+
+  depsSha256 = "0i1y57kv2yacw4ndqdc2pp56kvrd76fcwss8vqf2bh9whvdfqjfx";
+
+  buildInputs = [ makeWrapper ];
+
+  # for some reason RUST_SRC_PATH is not defined even if it's in preCheck
+  # let's just disable check
+  doCheck = false;
+
+  preInstall = ''
+    export RUST_SRC_PATH="${rustc.src}/src"
```

Comment:
All attributes in the `mkDerivation` call are passed to the build script via environment variables, so you can create an attribute `RUST_SRC_PATH = "${rustc.src}/src"` to achieve the same effect.